lib/auth/trustedcluster.go

```diff
@@ -367,14 +368,35 @@ func (a *AuthServer) updateRemoteClusterStatus(remoteCluster services.RemoteClus
 	if err != nil {
 		return trace.Wrap(err)
 	}
-	remoteCluster.SetConnectionStatus(teleport.RemoteClusterStatusOffline)
 	lastConn, err := services.LatestTunnelConnection(connections)
-	if err == nil {
-		offlineThreshold := time.Duration(keepAliveCountMax) * keepAliveInterval
-		tunnelStatus := services.TunnelConnectionStatus(a.clock, lastConn, offlineThreshold)
-		remoteCluster.SetConnectionStatus(tunnelStatus)
+	if err != nil {
+		if remoteCluster.GetConnectionStatus() != teleport.RemoteClusterStatusOffline {
+			remoteCluster.SetConnectionStatus(teleport.RemoteClusterStatusOffline)
+			if err := a.UpdateRemoteCluster(ctx, remoteCluster); err != nil {
+				return trace.Wrap(err)
+			}
+		}
+		return nil
+	}
```

Comment:
Nit: might be more future-proof to not categorically suppress errors from `services.LatestTunnelConnection`.  Maybe either only suppress if `trace.IsNotFound(err)`, or move this logic to before the call the `services.LatestTunnelConnection` and scope it to `if len(connections) == 0` instead.

Reply:
Makes sense, added a `trace.IsNotFound` check